src/fundus/scraping/html.py

```diff
@@ -107,12 +107,11 @@ def filter_url(u: str) -> bool:
             session = session_handler.get_session()
 
             try:
-                if self.query_parameters is not None:
-                    for key, value in self.query_parameters.items():
-                        if "?" in url:
-                            url += "&" + key + "=" + value
-                        else:
-                            url += "?" + key + "=" + value
+                for key, value in self.query_parameters.items():
+                    if "?" in url:
+                        url += "&" + key + "=" + value
+                    else:
+                        url += "?" + key + "=" + value
```

Comment:
Sorry, one last thing i missed: This should be outside the `try ... catch` block.